File: ontrack-repository-support/src/main/java/net/nemerosa/ontrack/repository/support/store/EntityDataStoreRecord.java
package net.nemerosa.ontrack.repository.support.store;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.Data;
import net.nemerosa.ontrack.model.structure.ProjectEntity;
import net.nemerosa.ontrack.model.structure.Signature;

@Data
public class EntityDataStoreRecord implements Comparable<EntityDataStoreRecord> {

    private final int id;
    private final ProjectEntity entity;
    private final String category;
    private final String name;
    private final String groupName;
    private final Signature signature;
    private final JsonNode data;

    @Override
    public int compareTo(EntityDataStoreRecord o) {
        int i = -this.signature.getTime().compareTo(o.signature.getTime());
        if (i != 0) {
            return i;
        } else {
            return o.id - this.id;
        }
    }
}

File: ontrack-repository-support/src/main/java/net/nemerosa/ontrack/repository/support/store/EntityDataStoreRecord.kt
package net.nemerosa.ontrack.repository.support.store

import com.fasterxml.jackson.databind.JsonNode
import net.nemerosa.ontrack.model.structure.ProjectEntity
import net.nemerosa.ontrack.model.structure.Signature

data class EntityDataStoreRecord(
    val id: Int,
    val entity: ProjectEntity?,
    val category: String,
    val name: String,
    val groupName: String?,
    val signature: Signature,
    val data: JsonNode
) : Comparable<EntityDataStoreRecord> {

    override fun compareTo(other: EntityDataStoreRecord): Int {
        val i = -this.signature.time.compareTo(other.signature.time)
        if (i != 0) {
            return i
        } else {
            return other.id - this.id
        }
    }

}
